metrics/registry.go

```diff
@@ -71,45 +77,60 @@ func (r *StandardRegistry) Each(f func(string, interface{})) {
 
 // Get the metric by the given name or nil if none is registered.
 func (r *StandardRegistry) Get(name string) interface{} {
-	r.mutex.Lock()
-	defer r.mutex.Unlock()
-	return r.metrics[name]
+	item, _ := r.metrics.Load(name)
+	return item
 }
 
 // Gets an existing metric or creates and registers a new one. Threadsafe
 // alternative to calling Get and Register on failure.
 // The interface can be the metric to register if not found in registry,
 // or a function returning the metric for lazy instantiation.
 func (r *StandardRegistry) GetOrRegister(name string, i interface{}) interface{} {
-	r.mutex.Lock()
-	defer r.mutex.Unlock()
-	if metric, ok := r.metrics[name]; ok {
-		return metric
+	// fast path
+	cached, ok := r.metrics.Load(name)
+	if ok {
+		return cached
 	}
 	if v := reflect.ValueOf(i); v.Kind() == reflect.Func {
 		i = v.Call(nil)[0].Interface()
 	}
-	r.register(name, i)
-	return i
+	if !r.isKnownType(i) {
+		return i
+	}
+	item, _ := r.register(name, i)
```

Comment:
Since you are no longer holding a mutex, you can wind up in the situation where you first do the `Load`, get nothing, and then go through this registering. Meanwhile, someone else may be registering it. Not sure how the new code would behave in that scenario.. ?

Reply:
If multiple goroutines manage to miss cache then they will behave as usual mutex protected map since LoadOrStore method uses inner mutex of sync.map for adding elements.